### PR TITLE
/access/: Add API to edit records

### PR DESCRIPTION
**Pull request by mistydemeo**

### CODE REVIEW

File: src/archivematicaCommon/lib/archivesspace/client.py [hunk 1]
```diff
@@ -130,6 +130,41 @@ def resource_type(self, resource_id):
     def get_record(self, record_id):
         return self._get(record_id).json()
 
+    def edit_record(self, new_record):
+        """
+        Update a record in ArchivesSpace using the provided new_record.
+
+        The format of new_record is identical to the format returned by get_resource_component_and_children and related methods; consult the documentation for that method in ArchivistsToolkitClient to see the format.
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivesspace/client.py:137]: Should this point to ATK, or ArchivesSpace (this file)?

File: src/archivematicaCommon/lib/archivistsToolkit/client.py [hunk 2]
```diff
@@ -7,6 +7,10 @@
 logger.addHandler(logging.NullHandler())
 
 
+class ArchivistsToolkitError(Exception):
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivistsToolkit/client.py:10]: Is this used anywhere?

File: src/dashboard/src/components/access/views.py [hunk 3]
```diff
@@ -81,15 +81,31 @@ def all_records(client, request, system=''):
 
 
 @_authenticate_to_archivesspace
-def get_record(client, request, system='', record_id=''):
-    record_id = record_id.replace('-', '/')
+def record(client, request, system='', record_id=''):
```
Comment (Hwesta) [inline src/dashboard/src/components/access/views.py:85]: Update the urls.py?

File: src/archivematicaCommon/lib/archivistsToolkit/client.py (hunk 2 above)
Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivistsToolkit/client.py:10]: Accidentally sliced out the place where it's used, added back.

File: src/dashboard/src/components/access/views.py (hunk 3 above)
Comment (mistydemeo) [inline src/dashboard/src/components/access/views.py:85]: Updated.

File: src/archivematicaCommon/lib/archivesspace/client.py (hunk 1 above)
Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivesspace/client.py:137]: ATK; the format is only defined in the docstring in that file.

File: src/dashboard/src/components/access/views.py [hunk 4]
```diff
@@ -81,15 +81,31 @@ def all_records(client, request, system=''):
...
+    if request.method == 'PUT':
+        try:
+            new_record = json.load(request)
+        except ValueError:
+            response = {
+                'success': False,
+                'message': 'No JSON object could be decoded from request body.',
+            }
+            return helpers.json_response(response)
```
Comment (Hwesta) [inline src/dashboard/src/components/access/views.py]: Should this return a non-200 status code?

Comment (mistydemeo) [inline src/dashboard/src/components/access/views.py]: Updated to 400.

File: src/dashboard/src/components/access/views.py [hunk 5]
```diff
@@ -81,15 +81,31 @@ def all_records(client, request, system=''):
...
+        })
+    elif request.method == 'GET':
+        record_id = record_id.replace('-', '/')
+
+        try:
+            records = client.get_resource_component_and_children(record_id,
+                                                                 recurse_max_level=3)
+            return helpers.json_response(records)
+        except ArchivesSpaceError as e:
+            return helpers.json_response({"error": True, "message": str(e)})
```
Comment (Hwesta) [inline src/dashboard/src/components/access/views.py]: Should this return a non-200 status code?  Also, here the response has `"error": True`, while above it has `'success': False` - make the API consistent?

File: src/archivematicaCommon/lib/archivesspace/client.py (hunk 1 above)
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivesspace/client.py:137]: Ok, good docstring then. :)

File: src/dashboard/src/components/access/views.py (hunk 5 above)
Comment (mistydemeo) [inline src/dashboard/src/components/access/views.py]: Updated to make it consistent.

**Comment by Hwesta** [comment]
:+1:

**Pull request merged**